ddns-scripts: better error management for gandi

LecrisUT opened the pull request:
Maintainer: @feckert / @Ansuel 

Description:
Added error checks similar to the ones used in `ddns-scripts-pdns` (#16429)

feckert commented:
Please remove the https://github.com/openwrt/packages/pull/16430/commits/4e9172f3fd3b61fe3b95c84a54c34debffdf51a the readybility script.
If you want to improve this then use `. /usr/share/libubox/jshn.sh` to build the json. You can then add that with json_dump.
See for example https://github.com/openwrt/openwrt/blob/ec6293febc244d187e71a6e54f44920be679cde4/package/network/utils/comgt/files/directip.sh#L93

LecrisUT commented:
Hopefully that the right approach, I'll test it once I'm at the router.

feckert commented on net/ddns-scripts/files/usr/lib/ddns/update_gandi_net.sh:
```diff
@@ -1,19 +1,37 @@
+json_init
+json_add_string rrset_ttl "$__TTL"
+json_add_array rrset_values "$__IP"
+ubus call network add_dynamic "$(json_dump)"
```
You could delete this line.
This is not necessary.
I just wanted to show you where you can have a look how to use jshn.

LecrisUT replied:
Oh, my bad, I thought I've deleted that part

feckert replied:
Please stash your commits to one after I have merged https://github.com/openwrt/packages/pull/16429 you should the bump the PKG_RELEASE number

LecrisUT replied:
Certainly, I will test these changes for myself as well and rebase all of the commits.
So the `PKG_RELEASE` should be bumped for both?

feckert replied:
> Certainly, I will test these changes for myself as well and rebase all of the commits.
> So the `PKG_RELEASE` should be bumped for both?
 
Yes we need a PKG_RELEASE bump for this pullrequest and for the other one.

> 
> Edit: BTW, what's with the CI error for `x86_64`?

I have asked myself the same question. I'll have to ask the others. I haven't looked at it yet.

LecrisUT commented:
Rebased this one on the `PowerDNS` one so we don't get a conflict later on. All changes are squashed now and everything has been tested.

I've also switched it to `LiveDNS API` since the old API has been/is being deprecated from conversations with Gandi team.
